core/src/main/java/com/oath/oak/OakWBufferImpl.java

```diff
@@ -14,15 +14,15 @@
 public class OakWBufferImpl implements OakWBuffer {
 
     private ByteBuffer bb;
-    private final ValueUtils valueOperator;
 
     OakWBufferImpl(Slice s, ValueUtils valueOperator) {
-        this.bb = s.getByteBuffer();
-        this.valueOperator = valueOperator;
-    }
+        ByteBuffer buff = s.getByteBuffer();
+        int initPos = buff.position();
+        buff.position(initPos + valueOperator.getHeaderSize());
+        this.bb = buff.slice();
```

Comment:
I am against using slice() method here. We had sliced the block's ByteBuffer previously and we intentionally removed this code, as it caused lots of small objects floating around for example in compute-heavy workload. We've seen big performance improvement due to this change. OakWBuffer is there to protect all the accesses. I would suggest to move the position and slice() only in getByteBuffer() method.